Improve readme (#91)
zeit/swr: README.md

Note that `fetcher` can be any asynchronous function, so you can use your favourite data-fetching
library to handle that part.

Check out [swr.now.sh](https://swr.now.sh) for more demos of SWR.

## Usage

Inside your React project directory, run the following:

```
yarn add swr
```

Or with npm:

```
npm install swr
```

### API

```js
const { data, error, isValidating, revalidate } = useSWR(key, fetcher, options)
```

> Example usage:

Reviewer: This seems out of place to me in the API section, can we move it down to the examples section under `### Local Configuration`?

Author: Hmmm actually I think this can be removed because we have the introduction section already.

> Example usage:

Author: ```suggestion
```

> ```js

Author: ```suggestion
```

> const { data, error } = useSWR('/api/user', fetcher, { refreshInterval: 3000 })

Author: ```suggestion
```

> ```

Author: ```suggestion
```

> which fetches data with `fetcher('/api/user')` every 3 seconds.

Author: ```suggestion
```

> 

Author: ```suggestion
```

#### Parameters

- `key`: a unique key string for the request (or a function / null) [(advanced usage)](#conditional-fetching)  
- `fetcher`: (_optional_) a Promise returning function to fetch your data [(details)](#data-fetching) 

> - `options`: (_optional_) options for this SWR hook

Reviewer: ```suggestion
- `options`: (_optional_) an object of options for this SWR hook
```

#### Return Values
- `data`: data for the given key resolved by `fetcher` (or undefined if not loaded)  
- `error`: error thrown by `fetcher` (or undefined)  
- `isValidating`: if there's a request or revalidation loading  
- `revalidate`: function to trigger the validation manually

#### Options

> - `suspense = false`: enabled React Suspense mode [(details)](#suspense-mode)

Reviewer: ```suggestion
- `suspense = false`: enable React Suspense mode [(details)](#suspense-mode)
```

- `fetcher = undefined`: the default fetcher function
- `revalidateOnFocus = true`: auto revalidate when window gets focused
- `refreshInterval = 0`: polling interval (disabled by default)
- `refreshWhenHidden = false`: polling when the window is invisible (if `refreshInterval` is enabled)
- `shouldRetryOnError = true`: retry when fetcher has an error [(details)](#error-retries)

> - `dedupingInterval = 2000`: deduping requests with the same key

Reviewer: ```suggestion
- `dedupingInterval = 2000`: dedupe requests with the same key in this time span
```
